sno/output_util.py

```diff
@@ -14,35 +21,48 @@ def dump_json_output(output, output_path, json_style="pretty"):
     Dumps the output to JSON in the output file.
     """
 
-    fp = resolve_output_path(output_path)
-
-    if json_style == 'pretty' and fp == sys.stdout and fp.isatty():
-        # Add syntax highlighting
-        from pygments import highlight
-        from pygments.lexers import JsonLexer
-        from pygments.formatters import TerminalFormatter
+    with resolve_output_path(output_path) as fp:
+        if json_style == 'pretty' and getattr(fp, 'supports_color', fp.isatty()):
+            # Add syntax highlighting
+            from pygments import highlight
+            from pygments.lexers import JsonLexer
+            from pygments.formatters import TerminalFormatter
 
-        dumped = json.dumps(output, **JSON_PARAMS[json_style])
-        highlighted = highlight(dumped.encode(), JsonLexer(), TerminalFormatter())
-        fp.write(highlighted)
-    else:
-        json.dump(output, fp, **JSON_PARAMS[json_style])
+            dumped = json.dumps(output, **JSON_PARAMS[json_style])
+            highlighted = highlight(dumped.encode(), JsonLexer(), TerminalFormatter())
+            fp.write(highlighted)
+        else:
+            json.dump(output, fp, **JSON_PARAMS[json_style])
 
 
-def resolve_output_path(output_path):
+@contextmanager
+def resolve_output_path(output_path, allow_pager=True):
```

Comment:
We should probably support [`core.pager`](https://git-scm.com/book/en/v2/Customizing-Git-Git-Configuration#_core_pager) from git config, maybe `=sno` and `=false`? I [turn it off in CI E2E tests](https://github.com/koordinates/sno/blob/f25eb993a68b3800fa28f7e3ddd3b4e24b7b84a4/tests/scripts/e2e-1.ps1#L56) since the CI envs have TTY things so output is pretty.